docs/develop/typescript/message-passing.mdx

```diff
@@ -196,107 +290,319 @@ await client.workflow.signalWithStart(yourWorkflow, {
+- `WorkflowExecutionUpdateCompleted` is added to the Event History when the Worker confirms that the Update has finished.
 
-### Define a Query {#define-query}
+To send an Update to a Workflow Execution, you can:
 
-A Query has a name and can have arguments.
+- Call [`WorkflowHandle.executeUpdate`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle/#executeUpdate) and wait for the Update to complete.
+  This code fetches an Update result:
 
-- The name, also called a Query type, is a string.
-- The arguments must be [serializable](/dataconversion).
+  ```typescript
+  let previousLanguage = await handle.executeUpdate(setLanguage, {
+    args: [Language.CHINESE],
+  });
+  ```
 
-Use [`defineQuery`](https://typescript.temporal.io/api/namespaces/workflow/#definequery) to define the name, parameters, and return value of a Query.
-This returns a `QueryDefinition` object.
-It carries the Query's signature.
-This includes the Query Type, the input parameters types, and the Query's return type.
+- Send [`WorkflowHandle.startUpdate`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle/#startUpdate) to receive an [`WorkflowUpdateHandle`](https://typescript.temporal.io/api/interfaces/client.WorkflowUpdateHandle) as soon as the Update is accepted or rejected.
 
-<!--SNIPSTART typescript-define-query -->
+  - Use this `UpdateHandle` later to fetch your results.
+  - `async` Update handlers normally perform long-running async activities.
+  - `startUpdate` only waits until the Worker has accepted or rejected the Update, not until all asynchronous operations are complete.
 
-[state/src/workflows.ts](https://github.com/temporalio/samples-typescript/blob/main/state/src/workflows.ts)
+  For example:
 
-```ts
-import { defineQuery } from '@temporalio/workflow';
+  ```typescript
+  const updateHandle = await handle.startUpdate(setLanguage, {
+    args: [Language.ENGLISH],
+    waitForStage: WorkflowUpdateStage.ACCEPTED,
+  });
+  previousLanguage = await updateHandle.result();
+  ```
 
-export const getValueQuery = defineQuery<number | undefined, [string]>(
-  'getValue',
-);
-```
+  For more details, see the "Async handlers" section.
 
-<!--SNIPEND-->
+To obtain an Update handle, you can:
 
-### Handle a Query {#handle-query}
+- Use [`WorkflowHandle.startUpdate`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle/#startUpdate) to start an Update and return the handle, as shown in the preceding example.
+- Use [`getUpdateHandle`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle/#getupdatehandle) to fetch a handle for an in-progress Update using the Update ID.
 
-Queries are handled by your Workflow.
 
-Don't include any logic that causes [Command](/workflows#command) generation within a Query handler (such as executing Activities).
-Including such logic causes unexpected behavior.
+:::info NON-TYPE SAFE API CALLS
```

Comment:
> NON-TYPE SAFE

I'M not sure about this, but shouldn't that be `NON-TYPE-SAFE`?

Reply:
Also, I know this is technically implied by "non type safe", but I would specifically call out that this call style results in loosing TypeScript's type safety checks on message arguments and return values.

Reply:
> I'M not sure about this, but shouldn't that be NON-TYPE-SAFE

I actually researched this recently... There are different opinions. One suggestion is to use an em-dash "Non—type-safe" :) One for the docs team, but I suspect @fairlydurable chose deliberately here.